core/spec.md

```diff
@@ -1178,98 +1205,129 @@ Regardless of how the model is retrieved, the overall format is as follows:
+    isn't already one defined for this level in the entity
   - Type: String
   - REQUIRED
+
 - `attributes."STRING".type`
   - The "TYPE" of the attribute being defined. MUST be one of the data types
     (in lower case) defined in [Attributes and
     Extensions](#attributes-and-extensions)
   - Type: TYPE
   - REQUIRED
+
 - `attributes."STRING".description`
   - A human readable description of the attribute
   - Type: String
   - OPTIONAL
+
 - `attributes."STRING".enum`
   - A list of possible values for this attribute. Each item in the array MUST
     be of type defined by `type`. When not specified, or an empty array, there
     are no restrictions on the value set of this attribute. This MUST only be
-    used when the `type` is a scalar. See the `strict`
-    attribute below
+    used when the `type` is a scalar. See the `strict` attribute below
   - Type: Array
   - OPTIONAL
+
 - `attributes."STRING".strict`
   - Indicates whether the attribute restricts its values to just the array of
     values specified in `enum` or not. A value of `true` means that any
     values used that is not part of the `enum` set MUST generate an error.
-    When not specified the default value is `true`. This attribute has no
-    impact when `enum` is absent or an empty array.
+    This attribute has no impact when `enum` is absent or an empty array
   - When not specified, the default value is `true`
   - Type: Boolean
   - OPTIONAL
+
 - `attributes."STRING".readonly`
-  - Indicates whether this attribute is a modifiable by a client. During
-    creation, or update, of an entity, if this attribute is specified then
-    its value MUST be ignored by the server even if the value is invalid.
-    When not specified the default value is `false. When the attribute name is
+  - Indicates whether this attribute is modifiable by a client.  During
+    creation, or update, of an entity if this attribute is specified then
+    its value SHOULD be ignored by the server even if the value is invalid.
+
+    Typically, attributes that are completely under the server's control
+    will be `readonly` - e.g. `self`.
+
+    When not specified the default value is `false`. When the attribute name is
     `*` then `readonly` MUST NOT be set to `true`. Note, both `clientrequired`
     and `readonly` MUST NOT be set to `true` at the same time
   - Type: Boolean
   - OPTIONAL
+
+- `attributes."STRING".immutable`
+  - Indicates whether this attribute's value can be changed once it is set.
+    Whether or not a client is allowed to set its initial value will be
+    controlled by the `readonly` attribute. The `id` attribute is an example
+    of an attribute that, once set, can not be changed.
+
+    Once set, any attempt to update the value MUST be silently ignored by
+    the server.
+
+    When not specified, the default value is `false. When the attribute name is
+    `*` then `immutable` MUST NOT be set to `true`. Note, both `clientrequired`
```

Comment:
I'd suggest moving "both" to just after "MUST NOT". (Otherwise having read the first part of the sentence, I'm expecting something that's independently true for both attributes, rather than both attributes *together*.)